mlflow/cli.py

```diff
@@ -281,6 +282,34 @@ def server(backend_store_uri, default_artifact_root, host, port,
         sys.exit(1)
 
 
+@cli.command()
+@click.option("--backend-store-uri", metavar="PATH",
+              default=DEFAULT_LOCAL_FILE_AND_ARTIFACT_PATH,
+              help="URI to which to persist experiment and run data. Acceptable URIs are "
+                   "SQLAlchemy-compatible database connection strings "
+                   "(e.g. 'sqlite:///path/to/file.db') or local filesystem URIs "
+                   "(e.g. 'file:///absolute/path/to/directory'). By default, data will be deleted "
+                   "from the ./mlruns directory.")
+@click.option("--run-ids", default=None,
+              help="Comma separated list of runs to be permanently deleted.")
+def gc(backend_store_uri, run_ids):
+    """
+    Permanently delete runs in state deleted in the backend store.
+    """
+    backend_store = _get_tracking_store(backend_store_uri, None)
+    if not run_ids:
+        run_ids = backend_store._get_deleted_runs()
```

Comment:
@t-henri I don't think we need a dedicated API for `_get_deleted_runs` that is defined on the backend stores. You can leverage the `run_view_type` parameter of the existing Search API (https://github.com/mlflow/mlflow/blob/9b73ef08c7bae1d49e1bb66d0c6aa8a5e2a0fe69/mlflow/store/tracking/abstract_store.py#L208) to fetch deleted runs.

Reply:
I thought about it but using the Search API is not convenient because it takes as parameter experiment_ids while I want deleted runs across all experiments. I could do the following things:
- Call list_experiments to get the list of all experiments and their experiment_ids
- Then use the Search API with these experiment_ids
But it seems not very efficient (issue multiple queries just to get the deleted runs) and will be more complicated code wise.
In my opinion implementing a simple _get_deleted_runs is much simpler, what do you think ?